### ISSUE TITLE
LLT doesn't detect backlit change after switching power mode

### ISSUE THREAD

**Issue by SharnavM** [BartoszCichecki/LenovoLegionToolkit#1436]
### Rules

- [X] I made myself familiar with the README.
- [X] I checked closed issues for similar ones and did not find a similar problem mentioned before.
- [X] I understand that, if insufficient information was provided, my issue will be closed without an answer.

### Version

2.24.1

### OS

Windows 11 Version 23H2 (OS Build 22631.4169)

### Device

Lenovo Legion Slim 7 16IAH7

### BIOS version

JYCN54WW

### What's wrong?

LLT doesn't take into account the backlit level when I switch power mode from Balanced to Quiet. When I switch the power mode to quiet, the backlit of my keyboard automatically turns off. LLT doesn't detect it and I have to press the `Fn + Spacebar` combo three times to bring it to the Initial level.

For Ex: Initially I have Keyboard backlit set to 'Low' and power mode is on 'Balanced'. After switching to 'Quiet' mode, keyboard backlit turns off automatically. Now the below happens:

- 1st key combo press: backlit is set to 'High'
- 2nd key combo press: backlit is 'Off'
- 3rd key combo press: backlit is set to 'Low'

### How to reproduce the bug?

1. Set power mode to 'Balanced'
2. Turn on the backlit
3. Switch to 'Quiet' mode
4. Press the key combo for backlit again

### What is the behavior that you expected?

I expect LLT to recognize that, I have switched to 'Quiet' mode and keyboard backlit has been turned off. It should do either of following:

1. Prevent the backlit from turning off and maintain its current level
2. Internally reset the backlit counter back to the "off" state. 

### Logs

[log_2024_09_12_09_55_46.txt](https://github.com/user-attachments/files/16977278/log_2024_09_12_09_55_46.txt)

### Additional information

https://github.com/user-attachments/assets/ef49d748-0402-4c11-a0f4-abf1a70b4c0c

**Comment by BartoszCichecki** [comment]
Unfortunately, this is how EC and keyboard controller behaves. When you switch to Quiet on battery, backlight will turn off but no notification is sent, so while this is annoying, it cannot be "fixed".

It would require Lenovo to update BIOS to fix this behaviour, because as it is now, when you switch to Quiet, keyboard still reports that the brightness is "Low" not "Off", even though it's actually off.

**Comment by SharnavM** [comment]
Thanks, I got it. Since it is a predefined behavior, I was curious if we could listen to power mode change, and use some if conditions to check if laptop is on battery and new power mode is 'Quiet' and re-set the backlit level to its current value (since keyboard doesn't report "off"). Or something like that?

**Comment by BartoszCichecki** [comment]
No, because it isn't predefined behaviour on all laptops. You can kind-of-workaround this with Actions, if you want.
